mitmproxy/tools/console/master.py

```diff
@@ -120,12 +121,33 @@ def get_editor(self) -> str:
         else:
             return "vi"
 
+    def get_hex_editor(self) -> str:
+        editors = [
+            "ghex",
+            "bless",
+            "xxd",
+            "imhex",
+            "hexedit",
+            "hexed",
+            "hxd",
+            "hexdump",
+            "emacs",
+        ]
+        for editor in editors:
+            if shutil.which(editor):
+                return editor
```

Comment:
Stupid question, but have you tested that these editors behave sanely? We need them to 

1. Open the binary when called with `$editor $filename`.
2. Keep the spawned process alive until the editor is closed.

I know that for text editors there are some offenders that spawn a GUI, but then return immediately. That would be a subpar user experience. :)

(Also let's add a comment in case people add other editors in the future)

Reply:
I confess I only tested ghex and bless, which were the ones I had installed. Others I included out of my head. I've just finished testing all the editors, except for HxD as I'm on Linux. Emacs happened exactly what you described, so I removed it. hexdump isn't really what I thought it was, and hexed is a browser editor, not a common windows editor as I thought (woops). xxd I included as IIRC was the default in ubuntu, but also doesn't work. I added a couple others I found that worked, including a vi one, but that I couldn't test too well as my brain can't process vim. I'll test some windows ones tomorrow.